### ISSUE TITLE
Singleplayer worlds with . character fail to load

### ISSUE THREAD

**Issue by TheMrEngMan** [mineblock11/continue-button#4]
If the world name does not match the world folder named saved on disk, it cannot be loaded into by pressing the continue button.

Example with `.` characters in name:
![image](https://github.com/mineblock11/continue-button/assets/68214507/ae5cc74c-14a3-48dc-8410-ee85c14d78a1)

The error when pressing continue:
![image](https://github.com/mineblock11/continue-button/assets/68214507/8a5ab1e6-b1c5-4484-aa4c-1b8e5c5d436d)

**Comment by TheMrEngMan** [comment]
The `1.2.2` version has un-done the partial fix implemented in version `1.2.1`.

The function [`removeIllegalCharacters` in version 1.2.1](https://github.com/mineblock11/continue-button/blob/1.2.1%2B1.20/src/main/java/com/mineblock11/continuebutton/ContinueButtonMod.java#L51) converted "1.20.1 Test" to "1_20_1_Test" (extra underscore).


The function [`removeIllegalCharacters` in version 1.2.2](https://github.com/mineblock11/continue-button/blob/1.2.2%2B1.20/src/main/java/com/mineblock11/continuebutton/ContinueButtonMod.java#L51) converted "1.20.1 Test" to "1.20.1 Test" (did not seem to do anything).

Proposed solution (basically [remove lines 61-64 from the version that was in 1.2.1](https://github.com/mineblock11/continue-button/blob/1.2.1%2B1.20/src/main/java/com/mineblock11/continuebutton/ContinueButtonMod.java#L61-L64)):

```java
public static final String removeIllegalCharacters(String name) {
    String safe = name.trim();
    safe = safe.replaceAll(
            "[\\/|\\\\|\\*|\\:|\\||\"|'|\\<|\\>|\\{|\\}|\\?|\\%|,]",
            "");
    safe = safe.replaceAll("\\.", "_");
    if (safe.endsWith("_")) {
        safe = safe.substring(0, safe.length() - 1);
    }
    return safe;
}
```

**Comment by TheMrEngMan** [comment]
@mineblock11 Actually, the above solution will only work for worlds which have not been significantly renamed. If a world is created and then renamed so that the world name and world folder name are completely different, the current implementation of using the world name will not work...
